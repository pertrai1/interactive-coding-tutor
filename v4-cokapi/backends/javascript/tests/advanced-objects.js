// Test advanced object features for Phase 3

// Object shorthand property notation
var name = "Alice";
var age = 30;
var person = {name, age};

// Computed property names
var key = "dynamicKey";
var obj = {[key]: "dynamicValue"};

// Method definitions
var calculator = {
  add(a, b) {
    return a + b;
  },
  multiply(x, y) {
    return x * y;
  }
};

var result = calculator.add(5, 3);
var product = calculator.multiply(4, 6);