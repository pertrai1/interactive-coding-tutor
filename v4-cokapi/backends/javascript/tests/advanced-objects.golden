{"code":"// Test advanced object features for Phase 3\n\n// Object shorthand property notation\nvar name = \"Alice\";\nvar age = 30;\nvar person = {name, age};\n\n// Computed property names\nvar key = \"dynamicKey\";\nvar obj = {[key]: \"dynamicValue\"};\n\n// Method definitions\nvar calculator = {\n  add(a, b) {\n    return a + b;\n  },\n  multiply(x, y) {\n    return x * y;\n  }\n};\n\nvar result = calculator.add(5, 3);\nvar product = calculator.multiply(4, 6);","trace":[{"line":4,"event":"step_line","func_name":"<module>","globals":{"name":"Alice"},"ordered_globals":["name"],"stack_to_render":[],"heap":{},"stdout":""},{"line":5,"event":"step_line","func_name":"<module>","globals":{"name":"Alice","age":30},"ordered_globals":["name","age"],"stack_to_render":[],"heap":{},"stdout":""},{"line":9,"event":"step_line","func_name":"<module>","globals":{"name":"Alice","age":30,"person":["INSTANCE","Object"],"key":"dynamicKey"},"ordered_globals":["name","age","person","key"],"stack_to_render":[],"heap":{},"stdout":""},{"line":20,"event":"step_line","func_name":"<module>","globals":{"name":"Alice","age":30,"person":["INSTANCE","Object"],"key":"dynamicKey","obj":["INSTANCE","Object"],"calculator":["INSTANCE","Object"]},"ordered_globals":["name","age","person","key","obj","calculator"],"stack_to_render":[],"heap":{},"stdout":""},{"line":22,"event":"step_line","func_name":"<module>","globals":{"name":"Alice","age":30,"person":["INSTANCE","Object"],"key":"dynamicKey","obj":["INSTANCE","Object"],"calculator":["INSTANCE","Object"],"result":8},"ordered_globals":["name","age","person","key","obj","calculator","result"],"stack_to_render":[],"heap":{},"stdout":""},{"line":23,"event":"step_line","func_name":"<module>","globals":{"name":"Alice","age":30,"person":["INSTANCE","Object"],"key":"dynamicKey","obj":["INSTANCE","Object"],"calculator":["INSTANCE","Object"],"result":8,"product":24},"ordered_globals":["name","age","person","key","obj","calculator","result","product"],"stack_to_render":[],"heap":{},"stdout":""}]}
